### PR TITLE
actual couch adding, now less WIP

### LINKED ISSUE #14: Host can add couch
As a user
when I visit my profile
and I click 'Host a Couch'
and I fill in the information about the couch
and I click 'List this Couch'
then I should be on the listing page for that couch
and I should see that the couch is unavailable
and I should see a link to "Add Availability" for my couch

### PR DESCRIPTION
**Pull request by Sh1pley**
@njgheorghita @riverswb @ski-climb @tmikeschu 

tried to grab all mentioned refactors and add them in.

for the new couch method in the couch controller, I could not get it to work without explicitly stating @user = User.find(params[:user_id]) 
would love suggestions or a fix on that, but that could be on a refactoring push.

closes #14

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class ApplicationController < ActionController::Base
   helper_method :current_user
 
   private
-  def current_user
-    @current_user ||= User.find(session[:current_user_id]) if session[:current_user_id]
-  end
+    def current_user
```
Comment (tmikeschu) [inline app/controllers/application_controller.rb:6]: Style points 👌

File: app/controllers/users/couches_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
+class Users::CouchesController < ApplicationController
```
Comment (tmikeschu) [inline app/controllers/users/couches_controller.rb:1]: I like how this reads!

File: app/views/users/couches/new.html.haml [hunk 3]
```diff
@@ -0,0 +1,28 @@
+%h1 Add Couch
```
Comment (tmikeschu) [inline app/views/users/couches/new.html.haml:1]: I'll make a card to add some html structure to this. Thanks for laying the groundwork!

File: app/views/users/couches/show.html.haml [hunk 4]
```diff
@@ -0,0 +1 @@
+= link_to('Add Availability', )
```
Comment (tmikeschu) [inline app/views/users/couches/show.html.haml:1]: To come with Brett's PR?

File: config/routes.rb [hunk 5]
```diff
@@ -3,6 +3,9 @@
   resources :sessions,  only: [:new]
   resources :users,     only: [:new, :create, :show] do
     scope module: :users do
+      resources :couches
+    end
+    scope module: :users do
       resources :reservations, only: [:index]
```
Comment (tmikeschu) [inline config/routes.rb:9]: Let's just throw this below line 6?

File: spec/features/users/user_can_list_a_couch_spec.rb [hunk 6]
```diff
@@ -0,0 +1,57 @@
+require 'rails_helper'
+
+describe 'User' do
```
Comment (tmikeschu) [inline spec/features/users/user_can_list_a_couch_spec.rb:3]: Very readable test!

File: spec/features/users/user_can_list_a_couch_spec.rb [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        fill_in "Name", with: "Bertha XXL"
+
+        fill_in "Street address", with: "123 IDK"
+        fill_in "City", with: "Some City"
+
+        fill_in "Zipcode", with: "11111"
+        click_on "Add Couch"
+      end
+
+      expect(current_path).to eq(new_user_couch_path(user, user.couches.new))
```
Comment (tmikeschu) [inline spec/features/users/user_can_list_a_couch_spec.rb:54]: Maybe expect to see an error message?

File: app/controllers/users/couches_controller.rb [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+
+  def new
+    @user = User.find(params[:user_id])
```
Comment (tmikeschu) [inline app/controllers/users/couches_controller.rb:4]: Are we able to just use `current_user` ?

**Comment by tmikeschu** [comment]
@Sh1pley stellar work. A couple suggestions/requests for refactoring, but I'm cool with merging this.

**Comment by njgheorghita** [comment]
I like it, thanks Jesse! Will merge so we can move ahead but let's keep the current_user issue in mind.

**Pull request merged**